Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/service/impl/ConceptMergeService.java

```diff
@@ -0,0 +1,170 @@
+    }
+
+    @Override
+    public void mergeConcepts(ConceptsMergeBean conceptsMergeBean, String userName)
+            throws LuceneException, IndexerRunningException, IllegalAccessException, DictionaryDoesNotExistException,
+            DictionaryModifyException {
+
+        if (conceptsMergeBean.getSelectedConceptId().trim().equals("")) {
+            // Add
+            ConceptEntry entry = new ConceptEntry();
+            createConceptEntryFromConceptsMergeBean(entry, conceptsMergeBean);
+            conceptManager.addConceptListEntry(entry, userName);
+        } else {
+            // Update
+            ConceptEntry entry = conceptManager.getConceptEntry(conceptsMergeBean.getSelectedConceptId());
+            createConceptEntryFromConceptsMergeBean(entry, conceptsMergeBean);
+            conceptManager.storeModifiedConcept(entry, userName);
+        }
+
+        deleteMergedConcepts(userName, conceptsMergeBean);
+
+    }
+
+    private void deleteMergedConcepts(String userName, ConceptsMergeBean conceptsMergeBean)
+            throws LuceneException, IndexerRunningException, IllegalAccessException,
+            DictionaryDoesNotExistException, DictionaryModifyException {
+        for (String conceptId : conceptsMergeBean.getConceptIds()) {
+            if (!conceptId.equalsIgnoreCase(conceptsMergeBean.getSelectedConceptId().trim())) {
+                if (ConceptTypes.SPECIFIC_WORDNET_CONCEPT == conceptTypesService
+                            .getConceptTypeByConceptId(conceptId)) {
+                    // If its a wordnet concept, then create a wrapper and set
+                    // the delete flag to true. This is done in two steps
+                    // because changeevent object needs to be updated for
+                    // deletion correctly.
+                    String conceptWrapperId = createConceptWrapperById(conceptId, userName, conceptsMergeBean);
+                    conceptManager.deleteConcept(conceptWrapperId, userName);
+                } else {
+                    // If its a CCP concept, just delete it.
+                    conceptManager.deleteConcept(conceptId, userName);
+                }
+            }
+        }
+    }
+
+    private String createConceptWrapperById(String conceptId, String userName, ConceptsMergeBean conceptsMergeBean)
+            throws IllegalAccessException, DictionaryDoesNotExistException, DictionaryModifyException, LuceneException,
+            IndexerRunningException {
+        ConceptEntry entry = conceptManager.getConceptEntry(conceptId);
+        // Creating concept wrapper with all the values, because in future we
+        // will be including manipulations on deleted wrappers as well.
+        createConceptEntryFromConceptsMergeBean(entry, conceptsMergeBean);
+        return conceptManager.addConceptListEntry(entry, userName);
+    }
+
+    public void createConceptEntryFromConceptsMergeBean(ConceptEntry entry, ConceptsMergeBean conceptMergeBean) {
```

Comment:
this is also not creating a new object (you're passing it as parameter). maybe 'fill' or 'transfer content' (then without the 'to concept entry to ...' to keep it short.